tutorial/source/svi_part_iii.ipynb

```diff
@@ -282,52 +323,51 @@
-    "            pyro.sample(\"obs\", dist.Bernoulli(...), obs=self.data)\n",
+    "            pyro.sample(\"obs\", dist.Bernoulli(f), obs=self.data)\n",
     "\n",
     "    def guide(self, use_decaying_avg_baseline):\n",
     "        # register the two variational parameters with pyro\n",
-    "        log_alpha_q = pyro.param(\"log_alpha_q\", self.log_alpha_q_0)\n",
-    "        log_beta_q = pyro.param(\"log_beta_q\", self.log_beta_q_0)\n",
-    "        alpha_q, beta_q = torch.exp(log_alpha_q), torch.exp(log_beta_q)\n",
+    "        alpha_q = pyro.param(\"alpha_q\", torch.tensor(self.alpha_q_0),\n",
+    "                             constraint=constraints.positive)\n",
+    "        beta_q = pyro.param(\"beta_q\", torch.tensor(self.beta_q_0),\n",
+    "                            constraint=constraints.positive)\n",
     "        # sample f from the beta variational distribution\n",
     "        baseline_dict = {'use_decaying_avg_baseline': use_decaying_avg_baseline,\n",
     "                         'baseline_beta': 0.90}\n",
     "        # note that the baseline_dict specifies whether we're using\n",
     "        # decaying average baselines or not\n",
-    "        pyro.sample(\"latent_fairness\", dist.Beta(alpha_q, beta_q),\n",
+    "        f=pyro.sample(\"latent_fairness\", NonreparameterizedBeta(alpha_q, beta_q),\n",
     "                    infer=dict(baseline=baseline_dict))\n",
     "\n",
-    "    def do_inference(self, use_decaying_avg_baseline, tolerance=0.05):\n",
+    "    def do_inference(self, use_decaying_avg_baseline, tolerance=0.80):\n",
     "        # clear the param store in case we're in a REPL\n",
     "        pyro.clear_param_store()\n",
-    "        # initialize the variational parameters for this run\n",
-    "        self.setup()\n",
     "        # setup the optimizer and the inference algorithm\n",
-    "        optimizer = optim.Adam({\"lr\": .0008, \"betas\": (0.93, 0.999)})\n",
+    "        optimizer = optim.Adam({\"lr\": .0005, \"betas\": (0.93, 0.999)})\n",
     "        svi = SVI(self.model, self.guide, optimizer, loss=\"ELBO\", trace_graph=True)\n",
     "        print(\"Doing inference with use_decaying_avg_baseline=%s\" % use_decaying_avg_baseline)\n",
     "\n",
-    "        # do up to 10000 steps of inference\n",
-    "        for k in range(10000):\n",
+    "        # do up to this many steps of inference\n",
+    "        for k in range(self.max_steps):\n",
     "            svi.step(use_decaying_avg_baseline)\n",
     "            if k % 100 == 0:\n",
     "                print('.', end='')\n",
     "                sys.stdout.flush()\n",
     "\n",
     "            # compute the distance to the parameters of the true posterior\n",
-    "            alpha_error = param_abs_error(\"log_alpha_q\", self.log_alpha_n)\n",
-    "            beta_error = param_abs_error(\"log_beta_q\", self.log_beta_n)\n",
+    "            alpha_error = param_abs_error(\"alpha_q\", self.alpha_n)\n",
```

Comment:
Can you confirm that this stopping criterion is still ok? I had to update some of the stopping criteria in tests after we switched from log-coordinates to linear coordinates.

Reply:
yes it's ok